yarn-project/sequencer-client/src/sequencer/sequencer.ts

```diff
@@ -691,31 +691,25 @@ export class Sequencer {
         };
       }),
       this.l2BlockSource.getL2Tips().then(t => t.latest),
-      this.p2pClient.getStatus().then(p2p => p2p.syncedToL2Block),
       this.l1ToL2MessageSource.getBlockNumber(),
     ] as const);
 
-    const [worldState, l2BlockSource, p2p, l1ToL2MessageSource] = syncedBlocks;
+    const [worldState, l2BlockSource, l1ToL2MessageSource] = syncedBlocks;
```

Comment:
I think we should still ask the P2P node if it's synched up the latest block otherwise the sequencer could start building a block before the P2P client had a chance to react to the `BlocksAdded` event.

Reply:
This is exactly what happened in the failing prover test!